### PR TITLE
[reddit] try_get larger thumbnail images with fallback

### PR DESCRIPTION
**Pull request by ozburo**
## Please follow the guide below

- You will be asked some questions, please read them **carefully** and answer honestly
- Put an `x` into all the boxes [ ] relevant to your *pull request* (like that [x])
- Use *Preview* tab to see how your *pull request* will actually look like

---

### Before submitting a *pull request* make sure you have:
- [ ] [Searched](https://github.com/ytdl-org/youtube-dl/search?q=is%3Apr&type=Issues) the bugtracker for similar pull requests
- [ ] Read [adding new extractor tutorial](https://github.com/ytdl-org/youtube-dl#adding-support-for-a-new-site)
- [ ] Read [youtube-dl coding conventions](https://github.com/ytdl-org/youtube-dl#youtube-dl-coding-conventions) and adjusted the code to meet them
- [ ] Covered the code with tests (note that PRs without tests will be REJECTED)
- [ ] Checked the code with [flake8](https://pypi.python.org/pypi/flake8)

### In order to be accepted and merged into youtube-dl each piece of code must be in public domain or released under [Unlicense](http://unlicense.org/). Check one of the following options:
- [x] I am the original author of this code and I am willing to release it under [Unlicense](http://unlicense.org/)
- [ ] I am not the original author of this code but it is in public domain or released under [Unlicense](http://unlicense.org/) (provide reliable evidence)

### What is the purpose of your *pull request*?
- [ ] Bug fix
- [x] Improvement
- [ ] New extractor
- [ ] New feature

---

### Description of your *pull request* and other information

The current RedditRIE extractor simply returns the root thumb url for thumbnails, which is very small. This improvement checks for a larger version and returns that instead.

### CODE REVIEW

**Comment by dstftw** [comment]
Extract all resolutions.

**Comment by ozburo** [comment]
You mean extract all the available images into a `thumbnails` key and leave the existing `thumbnail` as is? I see what you mean, that would actually be better!

**Comment by ozburo** [comment]
Yes, much nicer xD

File: youtube_dl/extractor/reddit.py [hunk 1]
```diff
@@ -118,11 +119,23 @@ def _real_extract(self, url):
         else:
             age_limit = None
 
+        thumbnails = []
+        images = try_get(
+            data, lambda x: x['preview']['images'][0]['resolutions']) or []
+        for image in images:
+            url = url_or_none(unescapeHTML(image['url']))
```
Comment (dstftw) [inline youtube_dl/extractor/reddit.py:126]: Breaks.

File: youtube_dl/extractor/reddit.py [hunk 2]
```diff
@@ -118,11 +119,23 @@ def _real_extract(self, url):
...
+            if url is not None:
+                thumbnails.append({
+                    'url': url,
+                    'width': int_or_none(image['width']),
+                    'height': int_or_none(image['height']),
```
Comment (dstftw) [inline youtube_dl/extractor/reddit.py:131]: Breaks.

File: youtube_dl/extractor/reddit.py [hunk 3]
```diff
@@ -118,11 +119,23 @@ def _real_extract(self, url):
         else:
             age_limit = None
 
+        thumbnails = []
+        images = try_get(
+            data, lambda x: x['preview']['images'][0]['resolutions']) or []
+        for image in images:
```
Comment (dstftw) [inline youtube_dl/extractor/reddit.py:125]: Breaks if `...['resolutions']` is not a `list`.

File: youtube_dl/extractor/reddit.py [hunk 4]
```diff
@@ -9,6 +9,7 @@
     float_or_none,
     try_get,
     url_or_none,
+    unescapeHTML,
```
Comment (dstftw) [inline youtube_dl/extractor/reddit.py:12]: Alphabetic.

**Review by dstftw — CHANGES_REQUESTED** [review]
Read coding conventions.

**Pull request closed without merging**